### PR TITLE
[otbn] Fix --help for otbn top sim

### PR DESCRIPTION
**Pull request by rswarbrick**

### CODE REVIEW

File: hw/ip/otbn/dv/verilator/otbn_top_sim.cc [hunk 1]
```diff
@@ -131,10 +132,15 @@ int main(int argc, char **argv) {
             << "==================" << std::endl
             << std::endl;
 
-  int ret_code = simctrl.Exec(argc, argv);
```
Comment (GregAC) [inline hw/ip/otbn/dv/verilator/otbn_top_sim.cc:134]: Can't we continue to use `VerilatorSimCtrl::Exec` with it's behaviour fixed for `--help`?

Comment (rswarbrick) [reply hw/ip/otbn/dv/verilator/otbn_top_sim.cc:134]: Well, the problem is that you need to "return" two things in that case: a return code plus a flag saying whether we actually ran a simulation. I guess we could change `VerilatorSimCtrl::Exec` to return a pair. Fortunately, the Ibex code that uses `VerilatorSimCtrl` already does this manually, so we can change the API without breaking it. Would you prefer that approach?

Comment (GregAC) [reply hw/ip/otbn/dv/verilator/otbn_top_sim.cc:134]: It seems a bit silly to have a copy of the body of the `Exec` function repeated here, I think we should alter `Exec` so it works for this use case. Alternatively add a 'WasSimulationRun' function to `VerilatorSimCtrl` so the `Exec` function retains the same interface which is used to determine if registers should be dumped and the ISS/RTL comparison done.

Comment (rswarbrick) [reply hw/ip/otbn/dv/verilator/otbn_top_sim.cc:134]: I much prefer the "more functional" style, so I've just pushed up a version that should look like that. What do you think?

**Review by GregAC — APPROVED** [review]
LGTM, thanks for the fix

**Pull request merged**